From 3e15abefe6733715acb8927c4f08c8b8a2f498b7 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Mon, 11 Jul 2022 14:02:14 +0200
Subject: [PATCH 0251/1141] remoteproc: tee: add firmware ID parameter on
 open_session.

To be able to retrieve instance associated to a firmware,
Add the firmware ID as parameter on open_session.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: Id65efa7202259dd91d6e0cf5c418c4d9dbe7aff0
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/265912
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 drivers/remoteproc/tee_remoteproc.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

--- a/drivers/remoteproc/tee_remoteproc.c
+++ b/drivers/remoteproc/tee_remoteproc.c
@@ -245,6 +245,7 @@ struct tee_rproc *tee_rproc_register(str
 {
 	struct tee_client_device *rproc_tee_device;
 	struct tee_ioctl_open_session_arg sess_arg;
+	struct tee_param param[MAX_TEE_PARAM_ARRY_MEMBER];
 	struct tee_rproc *trproc;
 	int ret;
 
@@ -266,9 +267,15 @@ struct tee_rproc *tee_rproc_register(str
 	 * TEE_IOCTL_LOGIN_REE_KERNEL?
 	 */
 	sess_arg.clnt_login = TEE_IOCTL_LOGIN_PUBLIC;
-	sess_arg.num_params = 0;
+	sess_arg.num_params = 1;
 
-	ret = tee_client_open_session(pvt_data.ctx, &sess_arg, NULL);
+	param[0] = (struct tee_param) {
+		.attr = TEE_IOCTL_PARAM_ATTR_TYPE_VALUE_INPUT,
+		.u.value.a = fw_id,
+	};
+
+
+	ret = tee_client_open_session(pvt_data.ctx, &sess_arg, param);
 	if (ret < 0 || sess_arg.ret != 0) {
 		dev_err(dev, "tee_client_open_session failed, err: %x\n",
 			sess_arg.ret);
